pkg/kubelet/kubelet.go

```diff
@@ -1916,25 +1906,28 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 	updated := false
 	for i := range node.Status.Conditions {
 		if node.Status.Conditions[i].Type == api.NodeReady {
-			newCondition.LastTransitionTime = node.Status.Conditions[i].LastTransitionTime
-			if node.Status.Conditions[i].Status != api.ConditionTrue {
-				kl.recordNodeOnlineEvent()
-			}
-			node.Status.Conditions[i] = newCondition
+			oldReadyNodeConditionStatus = node.Status.Conditions[i].Status
```

Comment:
Yeah, never mind. I think the only remaining logic problem is the one I just commented on.